project/src/pages/mainPage/MainPage.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import {Offer} from '../../types/offers';
+import Header from '../../components/header/Header';
+import Tabs from '../../components/tabs/Tabs';
+import OffersList from '../../components/offersList/OffersList';
+import MainPageEmpty from './MainPageEmpty/MainPageEmpty';
+
+type MainPageProps = {
+  offers: Offer[];
+}
+
+function MainPage({offers}: MainPageProps) {
+  if (!offers || offers.length <= 0) {
```

Comment:
Смотри, у тебя offers обязательный параметр и если его не передать, то у тебя будет ошибка TS еще на этапе компиляции. Поэтому можно смело заменить на 
```suggestion
  if (!offers.length) {
```